zookeeper-jute/src/main/java/org/apache/jute/BinaryOutputArchive.java

```diff
@@ -108,18 +116,22 @@ public void writeString(String s, String tag) throws IOException {
             return;
         }
         ByteBuffer bb = stringToByteBuffer(s);
-        writeInt(bb.remaining(), "len");
+        int strLen = bb.remaining();
+        writeInt(strLen, "len");
         out.write(bb.array(), bb.position(), bb.limit());
+        dataSize += strLen;
```

Comment:
Since we write an int first, share will `+4` here? Ditto `writeBuffer`.

Reply:
line_120. It will `+4` in `writeInt`.

Reply:
You're right 😆